conan/cli/commands/compare.py

```diff
@@ -0,0 +1,134 @@
+                        full_ref = ref
+                        matching_remote = remote
+                        break
+                except:
+                    continue
+            else:
+                try:
+                    latest_recipe_revision = conan_api.list.latest_recipe_revision(ref, remote)
+                except:
+                    continue
+                if full_ref is None or (latest_recipe_revision.timestamp > full_ref.timestamp):
+                    full_ref = latest_recipe_revision
+                    matching_remote = remote
+        if full_ref is None or matching_remote is None:
+            raise ConanException(f"No matching reference for {reference} in remotes.\n"
+                                 "If you want to check against a local recipe, add an additional --{old,new}-path arg.\n")
+
+        conan_api.download.recipe(full_ref, matching_remote)
+        cache_path = conan_api.cache.export_path(full_ref)
+        return full_ref, cache_path
+
+    def _export_recipe_from_path(path_to_conanfile, reference):
+        path = conan_api.local.get_conanfile_path(path_to_conanfile, cwd, py=True)
+        ref = RecipeReference.loads(reference)
+        export_ref, conanfile = conan_api.export.export(path=path,
+                                                        name=ref.name, version=ref.version,
+                                                        user=ref.user, channel=ref.channel,
+                                                        lockfile=None,
+                                                        remotes=enabled_remotes)
+        cache_path = conan_api.cache.export_path(export_ref)
+        return export_ref, cache_path
+
+    def _source(path_to_conanfile, reference):
+        if path_to_conanfile is None:
+            export_ref, cache_path = _download_ref_from_remote(reference)
+        else:
+            export_ref, cache_path = _export_recipe_from_path(path_to_conanfile, reference)
+        exported_path = conan_api.local.get_conanfile_path(cache_path, cwd, py=True)
+        conan_api.local.source(exported_path, name=export_ref.name, version=str(export_ref.version), user=export_ref.user,
+                               channel=export_ref.channel, remotes=enabled_remotes)
+        return export_ref, cache_path
+
+    old_export_ref, old_cache_path = _source(args.old_path, args.old_reference)
+    new_export_ref, new_cache_path = _source(args.new_path, args.new_reference)
+
+    ConanOutput().info(f"Generating diff from {old_export_ref.repr_notime()} to {new_export_ref.repr_notime()} (this might take a while)")
+    # TODO: This is internal, we should use the public API, but nothing exposes functionality like this
+    diff = _execute_command(f"git diff --no-index {old_cache_path} {new_cache_path}",
+                            # We ignore the errors because git diff returns 1 if there are differences
+                            ignore_error=True)
+
```

Comment:
```suggestion
    if args.old_path:
        conan_api.remove.recipe(old_export_ref)
    if args.new_path:
        conan_api.remove.recipe(new_export_ref)
```

So that non-realized revisions do not clobber the cache?